homeassistant/components/epson/__init__.py

```diff
@@ -76,3 +77,28 @@ async def async_unload_entry(hass: HomeAssistant, entry: ConfigEntry):
     if unload_ok:
         hass.data[DOMAIN].pop(entry.entry_id)
     return unload_ok
+
+
+async def async_migrate_entry(hass, config_entry):
+    """Migrate old entry."""
+    _LOGGER.debug("Migrating unique_id in %s", config_entry.version)
+
+    old_unique_id = config_entry.unique_id
```

Comment:
We didn't set a config entry unique_id earlier, right? The old unique id of the entities was the config entry entry_id. We can use this as a check to compare if we need to migrate entities but we can't use this to replace the old entity entry unique_id.

Reply:
should I use this - https://dev-docs.home-assistant.io/en/master/api/config_entries.html#homeassistant.config_entries.ConfigFlow.async_set_unique_id ?

Reply:
To set the config entry unique_id, use this:
https://github.com/home-assistant/core/blob/c9bdc9609cfeeff8328fb90b04e915c778419a09/homeassistant/config_entries.py#L927-L937

Note that config entry unique_id and entity unique_id are different things in the Home Assistant api. So we need to update both since this integration uses the same id for both.

Reply:
I'd do the entity entry migration here, but update the config entry in the media player platform.

Reply:
can you check if current commit?